.github/workflows/build-docs.yml

```diff
@@ -0,0 +1,44 @@
+name: Build Sphinx Documentation
+
+on:
+  push:
+    paths:
+      - "docs/**" # Only trigger when files in docs folder change
+    branches:
+      - "**"
+  pull_request:
+    paths:
+      - "docs/**" # Only trigger when files in docs folder change
+    branches:
+      - "**"
+
+jobs:
+  build-docs:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.x"
+          cache: "pip"
+
+      - name: Install dependencies
+        working-directory: ./docs
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+
+      - name: Build Documentation
+        working-directory: ./docs
+        run: |
+          make html
+
+      - name: Upload documentation artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: sphinx-docs
+          path: docs/_build/html/ # Updated path to reflect docs directory
+          retention-days: 7
```

Comment:
@activus-d don't we need to make the documentation available somewhere? Or did you consider that out of scope for this PR?

If it helps, I registered with ReadTheDocs and connected this repository, in case we wanted to host the docs with them.

Reply:
Sure, the docs would be deployed, and I can make a pull request or push a new commit to set up deployment to ReadTheDocs once you’re cool with the initial configuration and setup here. Let me know your thoughts!